api/install.go

```diff
@@ -4,48 +4,47 @@ import (
 	"encoding/json"
 	"net/http"
 
-	"github.com/gojekfarm/albatross/api/logger"
+	"github.com/gojekfarm/albatross/pkg/helmclient"
+	"github.com/gojekfarm/albatross/pkg/logger"
 )
 
-type InstallRequest struct {
-	Name      string                 `json:"name"`
-	Namespace string                 `json:"namespace"`
-	Chart     string                 `json:"chart"`
-	Values    map[string]interface{} `json:"values"`
-}
-
+// InstallResponse represents the API response to the install request
 type InstallResponse struct {
 	Error  string `json:"error,omitempty"`
 	Status string `json:"status,omitempty"`
+	Data   string `json:"data,omitempty"`
 }
 
 // Install return an http handler that handles the install request
-// TODO: we could use interface as well if everything's in same package
-func Install(svc Service) http.Handler {
+func Install() http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		defer r.Body.Close()
 
-		var req InstallRequest
-		if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		operation := helmclient.NewInstallOperation()
+		if err := json.NewDecoder(r.Body).Decode(operation); err != nil {
```

Comment:
I think we need to relook the status code and body population in case of errors entirely. The function for populating error response (respondError) does not allow custom status code.